build.gradle

```diff
@@ -320,6 +320,10 @@ project(":samza-yarn_$scalaVersion") {
     compile "org.scala-lang:scala-compiler:$scalaLibVersion"
     compile "org.codehaus.jackson:jackson-mapper-asl:$jacksonVersion"
     compile "commons-httpclient:commons-httpclient:$commonsHttpClientVersion"
+    compile "org.apache.httpcomponents:httpcore:$httpClientVersion"
+    compile "org.apache.httpcomponents:httpclient:$httpClientVersion"
+    compile "org.bouncycastle:bcpkix-jdk15on:$bouncyCastleVersion"
```

Comment:
Bouncycastle is an open source code which we can include for open source, and it is widely used for cryptography.

But anyway, I also talked with the security team which is also building a certificate signing service and they are also using bouncycastle as well.